internal/clients/http/azure/create_vm.go

```diff
@@ -38,24 +37,104 @@ const (
 	vmPollFrequency       = 10 * time.Second
 )
 
-func (c *client) CreateVM(ctx context.Context, location string, resourceGroupName string, imageName string, pubkey *models.Pubkey, instanceType clients.InstanceTypeName, vmName string, userData []byte) (*string, error) {
+func (c *client) BeginCreateVM(ctx context.Context, vmParams clients.AzureInstanceParams, vmName string) (string, error) {
```

Comment:
I see that you have removed vmName as well, should I remove it as well for consistency?

Reply:
Yeah, i guess so. That's specific to a single machine i feel like... But happy to discuss ofc :)